contracts/strategies/stargate/StargateStrategy.sol

```diff
@@ -0,0 +1,306 @@
+                        1
+                )
+        );
+        assert(
+            _STARGATE_ROUTER_SLOT ==
+                bytes32(
+                    uint256(
+                        keccak256("eip1967.strategyStorage.stargateRouter")
+                    ) - 1
+                )
+        );
+        assert(
+            _STARGATE_POOL_ID_SLOT ==
+                bytes32(
+                    uint256(
+                        keccak256("eip1967.strategyStorage.stargatePoolId")
+                    ) - 1
+                )
+        );
+        assert(
+            _STARGATE_REWARD_PID_SLOT ==
+                bytes32(
+                    uint256(
+                        keccak256("eip1967.strategyStorage.stargateRewardPid")
+                    ) - 1
+                )
+        );
+    }
+
+    function initializeStargateStrategy(
+        address _storage,
+        address _underlying,
+        address _vault,
+        address _rewardPool,
+        address _depositToken,
+        address _stargateRouter,
+        uint256 _stargatePoolId,
+        uint256 _stargateRewardPid
+    ) public initializer {
+        BaseUpgradeableStrategy.initialize(
+            _storage,
+            _underlying,
+            _vault,
+            _rewardPool,
+            usdc,
+            300, // profit sharing numerator
+            1000, // profit sharing denominator
+            true, // sell
+            1e18, // sell floor
+            12 hours // implementation change delay
+        );
+
+        require(
+            IStargateToken(_underlying).token() == _depositToken,
+            "underlying does not match deposit token"
+        );
+        require(
+            IStargateToken(_underlying).poolId() == _stargatePoolId,
+            "underlying does not match pool ID"
+        );
+
+        (address foundLpToken, , , ) = IStargateFarmingPool(_rewardPool)
+            .poolInfo(_stargateRewardPid);
+        require(
+            foundLpToken == _underlying,
+            "reward pool LP token does not match underlying"
+        );
+
+        _setDepositToken(_depositToken);
+        _setStargateRouter(_stargateRouter);
+        _setStargatePoolId(_stargatePoolId);
+        _setStargateRewardPid(_stargateRewardPid);
+    }
+
+    function depositArbCheck() external view returns (bool) {
+        return true;
+    }
+
+    function depositToken() public view returns (address) {
+        return getAddress(_DEPOSIT_TOKEN_SLOT);
+    }
+
+    function stargateRouter() public view returns (address) {
+        return getAddress(_STARGATE_ROUTER_SLOT);
+    }
+
+    function stargatePoolId() public view returns (uint256) {
+        return getUint256(_STARGATE_POOL_ID_SLOT);
+    }
+
+    function stargateRewardPid() public view returns (uint256) {
+        return getUint256(_STARGATE_REWARD_PID_SLOT);
+    }
+
+    function getRewardPoolValues() public returns (uint256[] memory values) {
```

Comment:
What is the use of this function?

Reply:
So this function is to get current pending STG amounts from the pool, it's mostly unused, just to see how much we can harvest, it can be view function I guess.
Do you wanna keep it or remove it?

Reply:
I think it can be removed. We won't use it. We estimate profit of doHardWork by simulating the doHardWork transaction.